packages/repository-tests/src/crud/relations/acceptance/has-many-inclusion-resolver.relation.acceptance.ts

```diff
@@ -0,0 +1,235 @@
+                // this helper should create the inclusion resolvers for us
+                ({customerRepo, orderRepo} = givenBoundCrudRepositories(
+                ctx.dataSource,
+                repositoryClass,
+                ));
+                // inclusionResolvers should be defined. And resolver for each relation should be
+                // create by the hasManyFactory at this point.
+                expect(customerRepo.inclusionResolvers).to.not.be.undefined();
+                expect(orderRepo.inclusionResolvers).to.not.be.undefined();
+                expect(customerRepo.orders.inclusionResolver).to.not.be.undefined();
+                expect(customerRepo.customers.inclusionResolver).to.not.be.undefined();
+                // inclusionResolvers shouldn't setup yet at this point
+                expect(customerRepo.inclusionResolvers).to.deepEqual(new Map());
+
+                customerRepo.inclusionResolvers.set('orders', customerRepo.orders.inclusionResolver);
+                customerRepo.inclusionResolvers.set('customers', customerRepo.customers.inclusionResolver);
+                await ctx.dataSource.automigrate([Customer.name, Order.name]);
+            }),
+            );
+
+            beforeEach(async () => {
+                await customerRepo.deleteAll();
+                await orderRepo.deleteAll();
+            });
+
+            it("defines a repository's inclusionResolvers property", () => {
+                expect(customerRepo.inclusionResolvers).to.not.be.undefined();
+                expect(orderRepo.inclusionResolvers).to.not.be.undefined();
+            });
+
+            it("throws an error if the repository doesn't have such relation names", async () => {
+                await orderRepo.create({
+                    customerId: existingCustomerId,
+                    description: 'Order from Order McForder, the hoarder of Mordor',
+                });
+                await expect(customerRepo.find({include: [{relation: 'managers'}]})
+                ).to.be.rejectedWith(
+                    `Invalid "filter.include" entries: {"relation":"managers"}`
+                );
+            });
+
+            it('throws error if the target repository does not have the registered resolver', async () => {
+                await orderRepo.create({
+                    customerId: existingCustomerId,
+                    description: 'Order from Order McForder, the hoarder of Mordor',
+                });
+                // unregister the resolver
+                customerRepo.inclusionResolvers.delete('orders');
+
+                await expect(customerRepo.find({include: [{relation: 'orders'}]})
+                ).to.be.rejectedWith(
+                    `Invalid "filter.include" entries: {"relation":"orders"}`
+                );
+                // reset
+                customerRepo.inclusionResolvers.set('orders', customerRepo.orders.inclusionResolver);
```

Comment:
If the assertion on lines 96-98 throws, then this line is never executed and the tests that follow will NOT have any resolver registered for `orders`.

I have two proposals for you:

1. Use a different name than `orders`, that way you don't have to touch `customerRepo.inclusionResolvers`
2. Move the code setting up `customerRepo.inclusionResolvers` from `before` to `beforeEach` hook. That way the resolvers are set up again before each test and you don't need to worry about resetting them in individual tests.